File: 4139-minimum-absolute-distance-between-mirror-pairs/minimum-absolute-distance-between-mirror-pairs.cpp
class Solution {
public:
    int minMirrorPairDistance(vector<int>& nums) {
        unordered_map<int,int>Map;
        int n=nums.size();
        int ans=INT_MAX;
        for(int i=0;i<n;i++){
            string s = to_string(nums[i]);
            reverse(s.begin(), s.end());
            int reversed_num = stoi(s);
            if(Map.find(nums[i])!=Map.end()){
                int idx=Map[nums[i]];
                ans=min(ans,abs(i-idx));
            }
            Map[reversed_num]=i;
        }
        return (ans==INT_MAX)?-1:ans;
    }
};

File: 4139-minimum-absolute-distance-between-mirror-pairs/minimum-absolute-distance-between-mirror-pairs.py
class Solution:
    def minMirrorPairDistance(self, nums: List[int]) -> int:
        Map=defaultdict(int)
        ans=float('inf')
        for idx,val in enumerate(nums):
            if val in Map:
                ans=min(ans,abs(idx-Map[val]))
            string=str(val)
            string=string[::-1]
            Map[int(string)]=idx

        if ans==float('inf'):
            return -1
        return ans          
             